chia/full_node/mempool_manager.py

```diff
@@ -583,7 +621,9 @@ def check_removals(
         # 1. Checks if it's been spent already
         for record in removals.values():
             if record.spent:
-                return Err.DOUBLE_SPEND, []
+                # Only consider it a double spend if this is not a fast forward
+                if record.name not in fast_forward_coin_ids:
```

Comment:
do you remember why we don't need to handle duplicate spends here too? If you have two spends that can be deduplicated, presumably they would  be considered a double-spend under normal circumstances too

Reply:
Yes: we don't allow any spend (even if it turns out to be eligible for deduplication) to spend an already spent coin. If we get two similar spends, that's a conflict, and we do handle the deduplication case at that level (see code below this line). The exception now is these specific singleton spends, so now we allow them to technically be in a temporary double spend situation here.